golos.publication/golos.publication.cpp

```diff
@@ -141,6 +141,12 @@ void publication::delete_post(account_name account, std::string permlink) {
 }
 
 void publication::upvote(account_name voter, account_name author, std::string permlink, int16_t weight) {
+    structures::post posttable_obj;
+    if (get_post(author, permlink, posttable_obj, NULL_LAMBDA))
```

Comment:
1. it's better to check in-place things like auth/battery/etc first and do db access (get_post, etc) after
2. set_vote already contains get_post, move this check there
3. make sure, post can be upvoted after closing